catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorTest.java

```diff
@@ -22,322 +9,195 @@
-        contentDirectoryMonitor.init();
+        ContentDirectoryMonitor monitor = createContentDirectoryMonitor();
+        submitConfigOptions(monitor, monitoredDirectoryPath, true, ATTRIBUTE_OVERRIDES);
         RouteDefinition routeDefinition = camelContext.getRouteDefinitions()
                 .get(0);
-        assertThat(routeDefinition.toString(), containsString(
-                "SetHeader[" + Constants.ATTRIBUTE_OVERRIDES_KEY + ", simple{Simple: test1=someParameter1,test2=someParameter2}"));
+        assertThat(routeDefinition.toString(),
+                containsString("SetHeader[" + Constants.ATTRIBUTE_OVERRIDES_KEY
+                        + ", simple{Simple: test1=someParameter1,test2=someParameter2}"));
     }
 
-    /**
-     * ********************************************************************************
-     */
-
-    private RouteDefinition createRouteWithAdvice(String monitoredDirectory,
-            boolean copyIngestedFiles) throws Exception {
-        camelContext = (ModelCamelContext) super.createCamelContext();
-        camelContext.start();
+    @Test
+    public void testRouteCreationMissingMonitoredDirectory() throws Exception {
+        ContentDirectoryMonitor monitor = createContentDirectoryMonitor();
+        submitConfigOptions(monitor, "", true);
+        assertThat(camelContext.getRouteDefinitions()
+                .size(), is(0));
+        assertTrue(monitor.getRouteDefinitions()
+                .isEmpty());
+    }
 
-        contentDirectoryMonitor = new ContentDirectoryMonitor(camelContext);
-        contentDirectoryMonitor.systemSubjectBinder = noOpProcessor;
-        contentDirectoryMonitor.setMonitoredDirectoryPath(monitoredDirectory);
-        contentDirectoryMonitor.setCopyIngestedFiles(copyIngestedFiles);
-
-        // Simulates what container would do once all setters have been invoked
-        contentDirectoryMonitor.init();
-
-        // Did not work because it expects the route to be "adviced" already exists. So the above
-        // init()
-        // call created the initial route with the "content:framework" node in it, and then this
-        // AdviceWithRouteBuilder replaced the content:framework with mock:result, but created a
-        // second
-        // route that had this change in it. (So still get NPE because first route fires and cannot
-        // resolve the "content" scheme)
-        camelContext.getRouteDefinitions()
-                .get(0)
-                .adviceWith(camelContext, new AdviceWithRouteBuilder() {
-                    @Override
-                    public void configure() throws Exception {
-                        // weave the node in the route which has id = content://framework
-                        // and replace it with the following route path
-                        weaveByToString(".*content:framework.*").replace()
-                                .to("mock:result");
+    private void doAndVerifyFileMove(File destinationFolder, File monitoredFolder,
```

Comment:
We have way too many wait loops and sleeps in this unit test. Since our goal is not to verify that the Camel route works as it's supposed to but just verify that the route that got created is properly created, we should try to leverage the `CamelTestSupport` functionality more so that we don't have to play with real files. I'm really worried that these tests will be a cross-platform headache and fail intermittently.

Let me know if you need help using the `CamelTestSupport`, I've used it elsewhere.